lib/cisco_node_utils/interface.rb

```diff
@@ -903,6 +928,442 @@ def switchport_trunk_native_vlan=(val)
       end
     end
 
+    # Interface private vlan configuration properties
+    # private vlan mode (host, promisc, trunk promisc , trunk sec)
+    # private vlan mapping
+    # private vlan native vlan
+    # private vlan allow vlan
+
+    def cli_error_check(result)
+      # The NXOS interface private vlan cli does not raise an exception
+      # in some conditions and instead just displays a STDOUT error message
+      # thus NXAPI does not detect the failure.
+      # We must catch it by inspecting the "body" hash entry
+      # returned by NXAPI. This vlan cli behavior is unlikely to change.
+      # Check for messages that can be safely ignored.
+
+      errors = /(ERROR:|VLAN:|Eth)/
+      return unless
+        result[1].is_a?(Hash) && errors.match(result[1]['body'].to_s)
+      # Split errors into a list, but keep the delimiter as part of the message.
+      error_list =
+        (result[1]['body'].split(errors) - ['']).each_slice(2).map(&:join)
+      error_list.each do |_msg|
+        fail result[1]['body']
+      end
+    end
+
+    def switchport_enable_and_mode_private_vlan_host(mode_set)
+      switchport_enable unless switchport
+      if mode_set[/(host|promiscuous)/]
+        config_set('interface', 'switchport_mode_private_vlan_host',
+                   name: @name, state: '', mode: IF_SWITCHPORT_MODE[mode_set])
+      else
+        config_set('interface', 'switchport_mode_private_vlan_host',
+                   name: @name, state: 'no', mode: IF_SWITCHPORT_MODE[mode_set])
+      end
```

Comment:
I still prefer having a single `config_set` and using variables for `state` and `mode`. 
But I grudgingly accept your approach.